### DISCUSSION TITLE
[WASM] ComboxBox + ControlTemplates: Selection is not highlighted

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#8524] [Q&A]
Please see attached: the selection on the dropdown is not highlighted
(follow up from #8515)

[DataTemplate3.zip](https://github.com/unoplatform/uno/files/8458283/DataTemplate3.zip)

**Comment by jeromelaban** [comment]
Thanks. If you change the `ItemContainerStyle` style and use the `DataTemplate` body to set the `Template` property, you can expect to lose all the default behavior for the `ComboBox` items.

You already set the `ComboBoxItemRevealStyle` (here's [the original template](https://github.com/unoplatform/uno/blob/e768418ed2381e34f69a6ba8ee340218b36de2e2/src/Uno.UI.FluentTheme.v2/Resources/Version2/Priority02/RevealBrush_rs4_themeresources.xaml#L407)), there should be no point in changing the `Template` for the `ComboBoxItem` as you're already changing the `ItemTemplate` on `ComboBox`.

It would be best to create the issues using known XAML syntax, for which you can find widely available samples to do what you need. Then you can port them to the C# only style you need at your convenience.

**Reply by DierkDroth** [reply]
Thanks again @jeromelaban 

Ok, the real problem is: I could not get the templates to work with the Uno.Material Theme (in fact my customized version of that theme) which is why I started messing with ItemContainerStyle in the first place.

- I confirm that removing the ControlTemplate from ItemContainerStyle works 100% correctly with Fluent style -> good
- however, I wanted to replicate the problem with Uno.Material - to demonstrate the original issue - [following guidelines here](https://platform.uno/docs/articles/features/uno-material.html#:~:text=Uno%20Material%20is%20an%20add,like%20Buttons%2C%20TextBox%2C%20etc.) and immediately ran into errors below.

Please find the corrupted repo below

```
Rebuild started...
1>------ Rebuild All started: Project: DataTemplate5.Windows, Configuration: Debug x86 ------
2>------ Rebuild All started: Project: DataTemplate5.Wasm, Configuration: Debug Any CPU ------
Restored C:\Users\dierk\Documents\repos\Uno\DataTemplate5\DataTemplate5.Wasm\DataTemplate5.Wasm.csproj (in 137 ms).
Restored C:\Users\dierk\Documents\repos\Uno\DataTemplate5\DataTemplate5.Windows\DataTemplate5.Windows.csproj (in 152 ms).
2>C:\Users\dierk\.nuget\packages\uno.ui\4.0.11\buildTransitive\netstandard2.0\Uno.UI.targets(12,3): error UNOB0001: Cannot build with both Uno.WinUI and Uno.UI nuget packages referenced.
2>Done building project "DataTemplate5.Wasm.csproj" -- FAILED.
1>C:\Users\dierk\Documents\repos\Uno\DataTemplate5\DataTemplate5.Shared\App.xaml(15,9): XamlCompiler error WMC0011: Unknown member 'ColorPaletteOverrideSource' on element 'MaterialColors'
1>Done building project "DataTemplate5.Windows.csproj" -- FAILED.
========== Rebuild All: 0 succeeded, 2 failed, 0 skipped ==========
```
[DataTemplate5.zip](https://github.com/unoplatform/uno/files/8462283/DataTemplate5.zip)

**Reply by kazo0** [reply]
It looks like your project is targeting WinUI so you would need to install the [Uno.Material.WinUI](https://www.nuget.org/packages/Uno.Material.WinUI) package instead of the Uno.Material package.

As for the second error, the `ColorPaletteOverrideSource` property on `MaterialColors` has been renamed to `OverrideSource`.

The documentation is in need of an update to address the split of Uno.UI/Uno.WinUI as well as the property name changes and introduction of the Uno.Toolkit.UI.Material library inside of the [Uno Toolkit](https://github.com/unoplatform/uno.toolkit.ui)

For now, a more updated doc would be the README from the [Uno Themes repo](https://github.com/unoplatform/Uno.Themes)

**Reply by DierkDroth** [reply]
@kazo0 thanks for follow up.

Unfortunately I still can't get it work after applying the changes you referenced. I get warnings below and an exception on startup. Updated repo attached

```
2>Uno.UI is using Roslyn generators
1>CSC : warning CS8785: Generator 'DependencyPropertyGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'Exception' with message 'Unable to find Uno.UI.Xaml.GeneratedDependencyPropertyAttribute'
1>CSC : warning CS8785: Generator 'XamlCodeGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentException' with message 'The path is not of a legal form.'
2>DataTemplate5.Wasm -> C:\Users\dierk\Documents\repos\Uno\DataTemplate5\DataTemplate5.Wasm\bin\Debug\net6.0\DataTemplate5.Wasm.dll
1>DataTemplate5.Windows -> C:\Users\dierk\Documents\repos\Uno\DataTemplate5\DataTemplate5.Windows\bin\x86\Debug\net6.0-windows10.0.19041.0\win10-x86\DataTemplate5.Windows.dll
1>WINAPPSDKGENERATEPROJECTPRIFILE : warning : PRI257: 0xdef00522 - Resources found for language(s) 'en' but no resources found for default language(s): 'en-US'. Change the default language or qualify resources with the default language. http://go.microsoft.com/fwlink/?LinkId=231899
1>Done building project "DataTemplate5.Windows.csproj".
```
[DataTemplate6.zip](https://github.com/unoplatform/uno/files/8463837/DataTemplate6.zip)

**Reply by jeromelaban** [reply]
Can you paste the exception you're getting? Make sure to post an runtime log, stack or build information in these cases.

**Reply by DierkDroth** [reply]
Sure

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/26179242/162740504-389b0e6d-a59a-4557-9a4c-e3e1a23280b6.png">

**Reply by kazo0** [reply]
Ah I see, it seems that you don't have a `ColorPaletteOverride.xaml` in your .Shared project. That file is only necessary if you wanted to override the default colors that come baked into the Uno Material library. So you will either need to add a `ColorPaletteOverride.xaml` (something like what's in the [Getting Started](https://github.com/unoplatform/Uno.Themes#getting-started) section of the Uno.Themes repo) or simply remove the `OverrideSource="ms-appx:///ColorPaletteOverride.xaml"` property setter from the `<MaterialColors .../>` line in your App.xaml.

**Reply by DierkDroth** [reply]
@kazo0 thanks that removed the exception.

However, now I'm still on on Fluent style (please see ComboBox dropdown which clearly is Fluent). How could I switch to Material to resume my testing/repro efforts?

**Reply by jeromelaban** [reply]
Can you try to reproduce using explicit XAML, just so that we're tracking an actual issue?

**Reply by DierkDroth** [reply]
@jeromelaban I wished I could. The problem is that I can't get the repo to do Material, it still does Fluent.

Please see attached:
- I tried to switch the repo to Material following guidelines above ->it still does Fluent (see styling of drop down)
- however, I moved the DataTemplate to XAML and loaded it from there -> works, but that never was my problem, since it always was working on Fluent

-> I needed to switch my repo to Material to reproduce the issue. What would I need to do?

[DataTemplate7.zip](https://github.com/unoplatform/uno/files/8465379/DataTemplate7.zip)

**Reply by jeromelaban** [reply]
Thanks. In the documentation, we explain that the default styles are not set by default unless you explicitly configure it.

If you set your styles this way, you'll get the behavior you need:
```xaml
<MaterialResources xmlns="using:Uno.Material" WithImplicitStyles="True" />
```

**Reply by DierkDroth** [reply]
@jeromelaban the docs stated "Optional" which is why I left that...

Anyway, I set that line in App.xaml (that the only change, that't why I held off uploading anotehr repo) and now am getting again an exception on startup (screenshot below). 

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/26179242/162781052-e280d01d-961d-4ab4-a061-32d3f30209f4.png">

**Reply by DierkDroth** [reply]
@jeromelaban please don't invest any more effort in this case: I'm about to reimplement the ComboBoxItem style for Material based on the reference you provided above. This looks promising so far as it resolves my original problem.

I'll get back to you in case I hit the wall again. Thanks for your support.
